File: 2020/interview/tpc/tpc-warmup-2-easy-number-game.py
def read_ints():
    return list(map(int, input().split(' ')))

def solve():
    t = int(input())
    for i in range(t):
        n, m = read_ints()
        nums = read_ints()
        nums.sort()
        l, r = 0, 2 * m - 1
        ret = 0
        while m and l < r:
            ret += nums[l] * nums[r]
            l += 1
            r -= 1
            m -= 1
        print(ret)

solve()

'''
3
4 2
1 3 2 4
3 1
2 3 1
4 0
1 3 2 4
'''

File: 2020/interview/tpc/warmup/tpc-warmup-2-easy-number-game.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main()
{
    int t;
    scanf("%d", &t);
    for (int i = 0; i < t; i += 1)
    {
        int n, m;
        scanf("%d %d", &n, &m);
        int* nums = new int[n];
        for (int j = 0; j < n; j += 1)
            scanf("%d", nums + j);
        sort(nums, nums + n);
        int l = 0, r = 2 * m - 1;
        long long int ret = 0;
        while (l < r)
        {
            ret += nums[l] * nums[r];
            l += 1, r -= 1;
        }
        printf("%lld\n", ret);
    }
}